redbot/core/core_commands.py

```diff
@@ -443,6 +468,407 @@ async def uptime(self, ctx: commands.Context):
+
+    # 1/30 minutes. It's not likely to change much and uploads a standalone webpage.
+    @commands.cooldown(1, 1800, commands.BucketType.user)
+    @mydata.command(cls=commands.commands._AlwaysAvailableCommand, name="3rdparty")
+    async def mydata_3rd_party(self, ctx: commands.Context):
+        """ View the End User Data statements of each 3rd-party module. """
+
+        # Can't check this as a command check, and want to prompt DMs as an option.
+        if not ctx.channel.permissions_for(ctx.me).attach_files:
+            ctx.command.reset_cooldown(ctx)
+            return await ctx.send("I need to be able to attach files (try in DMs?)")
+
+        statements = {
+            ext_name: getattr(ext, "__red_end_user_data_statement__", None)
+            for ext_name, ext in ctx.bot.extensions.items()
+            if not ext.__package__.startswith("redbot.")
+        }
+
+        if not statements:
+            return await ctx.send(
+                "This instance does not appear to have any 3rd party extensions loaded."
+            )
+
+        parts = []
+
+        formatted_statements = []
+
+        no_statements = []
+
+        for ext_name, statement in sorted(statements.items()):
+            if not statement:
+                no_statements.append(ext_name)
+            else:
+                formatted_statements.append(
+                    f"### {entity_transformer(ext_name)}" f"\n\n{entity_transformer(statement)}"
+                )
+
+        if formatted_statements:
+            parts.append(
+                "## 3rd party End User Data statements"
+                "\n\nThe following are statements provided by 3rd-party extensions."
+            )
+            parts.extend(formatted_statements)
+
+        if no_statements:
+            parts.append("## 3rd-party extensions without statements\n")
+            for ext in no_statements:
+                parts.append(f"\n - {entity_transformer(ext)}")
+
+        pretty_things = (  # This ends up being a small bit extra that really makes a difference.
+            """
+            <!DOCTYPE html>
+            <html>
+            <head><meta charset="utf-8">
+            <meta name="viewport" content="width=device-width, initial-scale=1">
+            <title>3rd Party Data Statements</title>
+            <style type="text/css">
+            body{margin:2em auto;max-width:800px;line-height:1.4;font-size:16px;
+            background-color=#EEEEEE;color:#454545;padding:1em;text-align:justify}
+            h1,h2,h3{line-height:1.2}
+            </style></head><body>
+            """
+        )
+
+        generated = markdown.markdown("\n".join(parts), output_format="html")
+
+        closing = "</body></html>"
+
+        html = "\n".join((pretty_things, generated, closing))
+
+        fp = io.BytesIO(html.encode())
+
+        await ctx.send(file=discord.File(fp, filename="3rd-party.html"))
+
+    async def get_serious_confirmation(self, ctx: commands.Context, prompt: str) -> bool:
```

Comment:
Does this return `None` on fail? I don't see any explicit `return False`s even though the return type is `bool`.